src/torch/0-tiefvision-commons/io/tiefvision_redis_io.lua

```diff
@@ -0,0 +1,88 @@
+-- Copyright (C) 2016 Pau Carré Cardona - All Rights Reserved
+-- You may use, distribute and modify this code under the
+-- terms of the Apache License v2.0 (http://www.apache.org/licenses/LICENSE-2.0.txt).
+
+--
+-- Reader and writer to store information thanks to redis
+--
+
+local paths = require('paths')
+local torchFolder = paths.thisfile('../..')
+package.path = string.format("%s;%s/?.lua", os.getenv("LUA_PATH"), torchFolder)
+
+local torch = require 'torch'
+local redis = require 'redis'
+local redisClient = nil
+
+local tiefvision_redis_io = {}
+function tiefvision_redis_io.read(fileName)
+  local key = toKey(fileName)
+  local response = tiefvision_redis_io.redisClient:hgetall(key)
+
+  local responseInt = {}
+  for key, value in pairs(response) do
+    local keyInt = tonumber(key)
+    responseInt[keyInt] = value
+  end
+
+  return responseInt
+end
+
+function tiefvision_redis_io.write(fileName, data)
+  local key = toKey(fileName)
+
+  local tmpFileName = paths.tmpname()
+  local file = io.open(tmpFileName, "w")
+
+  file:write(toRedisProtocol("DEL", key))
+  for i = 1, data:size()[1] do
+    file:write(toRedisProtocol("HSET", key, i, data[i]))
+  end
+
+  file:close()
+  os.execute("cat " .. tmpFileName .. " | redis-cli --pipe -h " .. tiefvision_redis_io.host .. " -p " .. tiefvision_redis_io.port .. " 1>/dev/null &")
+end
+
+function tiefvision_redis_io.last(fileName, data)
```

Comment:
The database can potentially grow and change over time, modifying these indexes.
I would rather pull all the product look keys from redis, make the difference with the product looks in the image folder and only compute distances with the new ones.
It'd be also necessary to keep a count of the images in the folder. In case the images in the image folder increases all the distances should be recomputed.